scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScaladocParser.scala

```diff
@@ -205,35 +205,37 @@ object ScaladocParser {
     P(startOrNl ~ (delimLine ~ nl).? ~ tableSpaceSep ~ table ~ (nl ~ delimLine).?)
   }
 
-  /** Contains all scaladoc parsers */
-  private val parser: Parser[Scaladoc] = {
-    val allParsers = Seq(
-      listBlockParser(),
-      codeBlockParser,
-      headingParser,
-      tagParser,
-      tableParser,
+  private def termParser[_: P] =
+    listBlockParser() |
+      codeBlockParser |
+      headingParser |
+      tagParser |
+      tableParser |
       leadTextParser // keep at the end, this is the fallback
-    )
-    val termParser = P(allParsers.reduce(_ | _))
-    val termEnd = End | paraEnd
-    val termsParser = P((!termEnd ~ termParser).rep(1))
-    val paraParser = termsParser.map(x => Scaladoc.Paragraph(x.toSeq))
-    val paraSep = (nl ~ &(nl)).rep(1)
-    val docParser = paraParser.rep(sep = paraSep).map(x => Scaladoc(x.toSeq))
+
+  // private def termParser[_: P] = P(allParsers.reduce(_ | _))
+
+  private def notTermEnd[_: P] = P(!(End | paraEnd))
+  private def termsParser[_: P] = P(notTermEnd ~ termParser)
+  private def paraParser[_: P] = P((termsParser.rep(1)).map(x => Scaladoc.Paragraph(x.toSeq)))
+  private def paraSep[_: P] = P((nl ~ &(nl)).rep(1))
+  private def docParser[_: P] = P(paraParser.rep(sep = paraSep).map(x => Scaladoc(x.toSeq)))
+
+  /** Contains all scaladoc parsers */
+  private def parser[_: P]: P[Scaladoc] = {
     P(paraSep.? ~ docParser ~ spacesMin(0) ~ End)
   }
 
   private val scaladocDelim = Pattern.compile("[ \t]*(?:$|\n[ \t]*\\**)")
 
   /** Parses a scaladoc comment */
-  def parse(comment: String): Option[Scaladoc] = {
+  def parseComment(comment: String): Option[Scaladoc] = {
```

Comment:
is this ok, renaming the public method?

Reply:
might not be, but didn't compile. I think I can work around it

Reply:
Reverted